Desktop_Interface/isobuffer.cpp

```diff
@@ -383,31 +393,44 @@ double isoBuffer::getDelayedTriggerPoint(double delay)
     auto isValid = [=](uint32_t index)->bool
     {
         if (m_back > delaySamples)
-            return (index < ((uint32_t)m_back - delaySamples)) || (index >= (uint32_t)m_back);
+            return (index < m_back - delaySamples) || (index >= m_back);
         else
-            // Fixme: There's probably an off by one here.
-            return (index < (m_bufferEnd + m_back - delaySamples)) && (index >= m_back);
+            return (index < m_bufferLen + m_back - delaySamples) && (index >= m_back);
     };
 
     auto getDelay = [=](uint32_t index)->double
     {
         if (m_back > index)
             return (m_back - index) / static_cast<double>(m_samplesPerSecond);
         else
-            return (m_bufferEnd + m_back - index) / static_cast<double>(m_samplesPerSecond);
+			// NOTE: Probably an off-by-one but I guess it's not too bad since it's being put into a floating point value?
+            return (m_bufferLen-1 + m_back - index) / static_cast<double>(m_samplesPerSecond);
     };
 
-    // Fixme: this won't look at the first element in the list.
-    for (auto it = std::prev(m_triggerPositionList.end()); it != m_triggerPositionList.begin(); it--)
-    {
-        if (isValid(*it))
-        {
-            uint32_t index = *it;
-            if (it != m_triggerPositionList.begin())
-                m_triggerPositionList.erase(m_triggerPositionList.begin(), std::prev(it));
-            return getDelay(index);
-        }
-    }
+	// Like std::find_if but returns the last element matching the predicate instead of the first one
+	// TODO: Move this elsewhere (maybe a utils / algorithms file??)
+	// requires first and last to be Bidirectional iters, and form a valid range
+	// requires p to be a valid unaryPredicate
+	// Full signature would be:
+	// template<typename It, typename Predicate>
+	// It find_last_if(It begin, It end, Predicate p)
+	auto find_last_if = [](auto begin, auto end, auto p)
```

Comment:
This is cool.  :)

Reply:
Thank you! I was gonna use reverse iterators originally but it was confusing to me, and converting reverse iterators to regular iterators to pass to vector::erase was messy, so i just wrapped it in an STL style algorithm.